fooof/objs/fit.py

```diff
@@ -732,7 +765,24 @@ def set_check_data_mode(self, check_data):
             Whether to run in check data mode.
         """
 
-        self._check_data = check_data
+        self.set_check_modes(check_data=check_data)
+
+
+    def set_run_modes(self, debug, check_freqs, check_data):
+        """Simultaneously set all run modes.
+
+        Parameters
+        ----------
+        debug : bool
+            Whether to run in debug mode.
+        check_freqs : bool
+            Whether to run in check freqs mode.
+        check_data : bool
+            Whether to run in check data mode.
+        """
+
+        self.set_debug_mode(debug)
+        self.set_check_modes(check_freqs, check_data)
```

Comment:
`set_check_modes`, `set_debug_mode`, and `set_run_modes` have overlap. it may be worth consolidating these into just `set_run_modes` since all functionality is contained there, while keeping the api simpler to a single method.

Reply:
Yeh... I agree that it's feeling a bit like overkill right now. For 1.1, since `set_debug_mode` is already there, I don't want to make a breaking change to remove it, and so it seemed most consistent to end up with also adding `set_check_modes`. 

However, for 2.0, when we have more leeway for changing things, I am thinking maybe drop the extras and consolidate to just `set_run_modes` - do you think it makes sense to do that update there?